spark/v3.1/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -115,6 +121,20 @@ public String wapId() {
     return sessionConf.get("spark.wap.id", null);
   }
 
+  public int outputSpecId() {
+    final int outputSpecId =
```

Comment:
Nit: Would you mind remove 'final' here?  I know from  @aokolnychyi (when he reviewed me change), he prefers not to have extra finals except in class fields, as modern compiler usually adds it anyway.  ex: https://github.com/apache/iceberg/pull/4293#discussion_r823011381

Reply:
Done. I usually just keep the final as a way of having something like `const` to avoid accidentally modifying something I did not intend to. But I think it is not necessary in this case and would prefer to do it to keep consistency in the rpo.